### PR TITLE
Feature/add dnsconfig dnspolicy

### PR DESCRIPTION
**Pull request by shinji62**
Please check [IDEA-106](https://yugabyte.atlassian.net/jira/polaris/projects/IDEA/ideas/view/2813870?selectedIssue=IDEA-106) for details.

But in short, adding the feature to be able to specify dnsConfig / dnsPolicy

### CODE REVIEW

File: stable/yugabyte/templates/service.yaml [hunk 1]
```diff
@@ -323,6 +323,12 @@ spec:
         {{- $_ := set $root.Values.tserver.affinity "podAntiAffinity" $podAntiAffinityData -}}
         {{ toYaml $root.Values.tserver.affinity | nindent 8 }}
         {{ end }}
+      {{- if $root.Values.dnsConfig }}
+      dnsConfig: {{- toYaml $root.Values.dnsConfig | nindent 8 }}
+      {{- end }}
```
Comment (bhavin192) [inline stable/yugabyte/templates/service.yaml]: We can use with here as well. Same for other occurrences. 
```suggestion
      {{- with $root.Values.dnsConfig }}
      dnsConfig: {{- toYaml . | nindent 8 }}
      {{- end }}
```

Comment (shinji62) [reply stable/yugabyte/templates/service.yaml]: Updated with changes @bhavin192

**Review by amannijhawan — APPROVED** [review]
Change looks good to me.
Not sure if we want to separate out dnsConfig of master and tserver but other than that change looks good.

**Comment by shinji62** [comment]
@amannijhawan Same settings are ok for both master and tserver

**Pull request merged**